app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.view
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * A page for displaying Onboarding Content.
+ *
+ * @param onboardingPageState The page content that's displayed.
+ * @param onPrimaryButtonClick Invoked when the user clicks the primary button.
+ * @param onSecondaryButtonClick Invoked when the user clicks the secondary button.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun OnboardingPage(
+    onboardingPageState: OnboardingPageState,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier.semantics(mergeDescendants = true) {},
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.SpaceEvenly,
```

Comment:
**This is a blocking change**: This is unfortunately gonna change the layout from the designs in https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106?node-id=6058%3A209483&t=bMa3YpVzO7xeRq9s-0. In particular, there's too much space between the Buttons and the Indicator when we would ideally want the Buttons to be at the bottom. We should ideally not change the layout from the desired specs in this refactoring.

Before (left) vs After (right):
<img width="1140" alt="Screen Shot 2023-01-05 at 10 10 02 PM" src="https://user-images.githubusercontent.com/1190888/210922275-89a2ece9-132a-4369-bc8c-3a983693207c.png">

Reply:
Umm I think this specific preview size works best for the left option but a different preview size might work better for the option on the right. At this moment, none of these are ideal, but as we discussed in the [dimensions + box with constraints PR](https://github.com/mozilla-mobile/fenix/pull/28348) and slack, that we could use grids and ratios to solve the problem of supporting different screen sizes. Adding a hard coded dimension here to make the button move down would look good in our previews but won't on different screen sizes. If you have a concrete solution, I'm open.

Reply:
One approach would be to pursue the original `SpaceBetween` approach with the 3 containers being:
- Dismiss button
- Image and text
- Buttons and Indicator
I do understand this change will likely break some of the foundations how you have chosen to refactor each section.

The spacing between the Button and Indicator does fall between 24dp and 32dp depending on whether or not the Secondary Button is shown. Since 24 and 32 are actual spacing values that we can find in the [Spacing Spec](https://www.figma.com/file/pEyGeE4KV5ytYHeXMfLcEr/Mobile-Styles?node-id=1017%3A56444&t=MxgC2NnW8WhXmN0q-0), we should treat the spacing here as intentional. Using `SpaceEvenly` would not satisfy this requirement because the buttons would be anchored to the Column with image and text, and disconnected from the indicators. There's also 16dp spacer between the button and indicator in [OnboardingContent](https://github.com/mozilla-mobile/fenix/pull/28376/files#diff-cee11c3fc1e078c563b8bc30d76866b9f01ead3fe2f7e1deebf70e271744cf6dR161) so I would imagine the spacing between the button and indicator would never follow the spec even if we had a grid system. 

While I do understand what you're saying, we can't simply just deviate from the original design spec for this reasoning alone since we have already gotten prior UX review for this. 

My ask here is that we shouldn't change the layout from the status quo in a refactoring. If we want to address the layout and scaling issues, let's address that separately or fix it before this refactoring so we are better guided on how to best refactor this. The underlying problem that I still want to focus on is only the spacing between the indicators and buttons.

Reply:
Hmm, updated it, could you take a look again?

Reply:
I discussed what I was still seeing with @rocketsroger and @MozillaNoah. There are still fundamental issues with how things are laid out that are breaking the design specs with the current approach. I would still suggest laying things out based on how I have described it and can be seen in the original implementation:

Column [Space Between]:
  - Dismiss button
  - Image and text
  - Buttons and Indicator

In your current implementation, we can't use `SpaceBetween` between the image and text because the spacing between them will now be dynamic. The image and text should be in the same container with a 32dp spacer between them to match the designs:
<img width="250" alt="Screen Shot 2023-01-09 at 5 38 23 PM" src="https://user-images.githubusercontent.com/1190888/211422466-1b52db73-1d45-4b15-912c-34aca5d3b30d.png">

The spacing between the button and indicators are also dependent on how many buttons are shown:
| One Button | Two Buttons |
|---|---|
|<img width="739" alt="Screen Shot 2023-01-09 at 5 41 46 PM" src="https://user-images.githubusercontent.com/1190888/211423543-82801a8e-f6e9-4977-8d8c-cc34fb8d98e9.png">|<img width="735" alt="Screen Shot 2023-01-09 at 5 41 51 PM" src="https://user-images.githubusercontent.com/1190888/211423548-ede58aca-a90e-4bb6-8ae5-e66ee96fb1f0.png">|

The `Spacer` as the first item in this `SpaceBetween` Column is also a bit of a bandaid solution that is masking other problems, which I don't really recommend since the intent of the refactoring was to make all of this better. We have already drastically changed how items are wrapped in their various column containers and as a result the layout has also changed. We should proceed somewhat cautiously to avoid further review churn. I am aware this is high priority work, but my understanding from @rocketsroger was that there is still a path forward that does not depend on this refactoring.